index.d.ts

```diff
@@ -1,12 +1,49 @@
-import { Parser } from 'acorn' 
+import * as acorn from 'acorn';
 
-declare const jsx: (options?: jsx.Options) => (BaseParser: typeof Parser) => typeof Parser;
+export const jsx: (options?: jsx.Options) => (BaseParser: typeof acorn.Parser) => typeof AcornJsxParser
 
-declare namespace jsx {
-  interface Options {
-    allowNamespacedObjects?: boolean;
-    allowNamespaces?: boolean;
-  }
+export interface Options {
+  allowNamespacedObjects?: boolean;
+  allowNamespaces?: boolean;
 }
 
-export = jsx;
+export const tokTypes: {
+  jsxName: acorn.TokenType,
+  jsxText: acorn.TokenType,
+  jsxTagEnd: acorn.TokenType,
+  jsxTagStart: acorn.TokenType
+} & typeof acorn.tokTypes;
+
+export const tokContexts: {
+  tc_oTag: acorn.TokContext,
+  tc_cTag: acorn.TokContext,
+  tc_expr: acorn.TokContext
+};
+
+export class AcornJsxParser extends acorn.Parser {
```

Comment:
This declares `AcornJsxParser` is exported as named export, and that it can be contructed from that import.

Reply:
I've added a commit to avoid doing so, while still ensuring we are copying over acorn's static `acorn` property. To do all this it seems to have been necessary to have separate constructor and instance interfaces.